sw/device/tock/dif/src/lib.rs

```diff
@@ -0,0 +1,14 @@
+// Copyright lowRISC contributors.
```

Comment:
Is this library *just* generated binding symbols? If so, I would consider renaming it `dif_bindings` to make it clear that it does not export a nice Rust API.

Reply:
@mcy thank you for reviewing this, I will try to address these comments tomorrow.

> Throughout: please document every function and variable, as we would in C. Tock has a bad habit of not doing this, but I don't think that's reason to not write the comments.

90% of the code is not written by me, I think the cleanup can be done later in follow-up PRs. At this point most of the code has been taken from the upstream, and is expected to gradually divert.